### PR TITLE
Actor State `ttlExpireTime`: Pipe `ttlExpireTime` to state responses

### PR DESCRIPTION
**Pull request by JoshVanL**
Part of https://github.com/dapr/dapr/issues/5899

PR pipes `ttlExpireTime` to get state responses, including actors.

Adds integration tests for state and actor state TTLs. Updates the existing actor state e2e test to check for `ttlExpireTime`.

Fixed an issue where an actors go routine was not returning on close.

### CODE REVIEW

File: tests/dapr_tests.mk [hunk 1]
```diff
@@ -90,7 +90,7 @@ DAPR_CONTAINER_LOG_PATH?=./dist/container_logs
 DAPR_TEST_LOG_PATH?=./dist/logs
 
 ifeq ($(DAPR_TEST_STATE_STORE),)
-DAPR_TEST_STATE_STORE=redis
+DAPR_TEST_STATE_STORE=postgres
```
Comment (ItalyPaleAle) [inline tests/dapr_tests.mk:93]: Let's keep this unrelated change in a different PR please.

**Review by ItalyPaleAle — DISMISSED** [review]

File: tests/dapr_tests.mk (hunk 1 above)
Comment (JoshVanL) [reply tests/dapr_tests.mk:93]: I still need to finalize this PR, but this is because redis doesn't support `ttlExpireTime`.

File: pkg/actors/placement/placement.go [hunk 2]
```diff
@@ -314,7 +315,7 @@ func (p *actorPlacement) establishStreamConn(ctx context.Context) (established b
 		}
 
 		err := p.client.connectToServer(ctx, serverAddr)
-		if err == errEstablishingTLSConn {
```
Comment (ItalyPaleAle) [inline pkg/actors/placement/placement.go:317]: Changes to this file seem unrelated

File: pkg/grpc/api_actor_test.go [hunk 3]
```diff
@@ -134,6 +134,9 @@ func TestGetActorState(t *testing.T) {
 			Key:       "key1",
 		}).Return(&actors.StateResponse{
 			Data: data,
+			Metadata: map[string]string{
+				"ttlExpireTime": "2020-10-02T22:00:00Z",
```
Comment (artursouza) [inline pkg/grpc/api_actor_test.go:138]: Sorry to be that person, but why not just "expiration"? Is "ttlExpireTime" used in other systems and we want to mimic?

File: pkg/actors/actors.go [hunk 4]
```diff
@@ -287,9 +287,9 @@ func (a *actorsRuntime) startAppHealthCheck(ctx context.Context, opts ...health.
 	for {
 		select {
 		case <-ctx.Done():
-			break
+			return
```
Comment (artursouza) [inline pkg/actors/actors.go:290]: Why is this needed?

File: tests/config/dapr_postgres_state.yaml [hunk 5]
```diff
@@ -20,10 +20,10 @@ spec:
...
   initTimeout: 1m
   metadata:
-  - name: connectionString
-    value: "host=dapr-postgres-postgresql.dapr-tests.svc.cluster.local user=postgres password=example port=5432 connect_timeout=10 database=dapr_test"
-  - name: table
-    value: querytable
+    - name: connectionString
+      value: "host=dapr-postgres-postgresql.dapr-tests.svc.cluster.local user=postgres password=example port=5432 connect_timeout=10 database=dapr_test"
+    - name: table
+      value: statetable
```
Comment (artursouza) [inline tests/config/dapr_postgres_state.yaml:26]: Why this change is needed?

**Review by artursouza — CHANGES_REQUESTED** [review]
LGTM overall but have some questions, mainly about naming. This is a contract after being merged.

File: pkg/grpc/api_actor_test.go (hunk 3 above)
Comment (JoshVanL) [reply pkg/grpc/api_actor_test.go:138]: `expiration` could be ambiguous. It is clearer that `ttlExpireTime` refers to and pairs with the `ttl` field.

File: pkg/actors/actors.go (hunk 4 above)
Comment (JoshVanL) [reply pkg/actors/actors.go:290]: The logic is currently bugged. The `break` statement will break from the `select` block and not the `for` loop.

File: tests/config/dapr_postgres_state.yaml (hunk 5 above)
Comment (JoshVanL) [reply tests/config/dapr_postgres_state.yaml:26]: It doesn't matter all that much, but the table name is conflicting with the [query state store](https://github.com/dapr/dapr/blob/e57a91fc8d2930eca71c8fdbf20084ed18161fc2/tests/config/dapr_postgres_query_state.yaml#L26).

**Comment by JoshVanL** [comment]
Flake?

File: pkg/grpc/api_actor_test.go (hunk 3 above)
Comment (artursouza) [reply pkg/grpc/api_actor_test.go:138]: "ttl" is a duration: "1min" or "3 hours". In this case, it means the expiration timestamp, like the date we see stamped at food labels.

File: pkg/actors/actors.go [hunk 6]
```diff
@@ -660,7 +660,8 @@ func (a *actorsRuntime) GetState(ctx context.Context, req *GetStateRequest) (*St
 	}
 
 	return &StateResponse{
-		Data: resp.Data,
+		Data:     resp.Data,
+		Metadata: resp.Metadata,
```
[the remark this answers is not in the document]
Comment (JoshVanL) [reply pkg/actors/actors.go:664]: It is currently only `ttlExpireTime`.

File: pkg/grpc/api_actor_test.go (hunk 3 above)
Comment (JoshVanL) [reply pkg/grpc/api_actor_test.go:138]: Yes, `ttl` is what is given in the metadata when saving and is a duration. `ttlExpireTime` is on the get response and is the timestamp of when that `ttl` duration will take effedct.

Comment (mukundansundar) [reply pkg/grpc/api_actor_test.go:138]: Can we shorten it? `ttlExpires` similar to HTTP `Expires` header but more qualified to `ttl` expiry. Wdyt?

Comment (JoshVanL) [reply pkg/grpc/api_actor_test.go:138]: It will need to be changed on the contrib side, as [that is where it is coming from](https://github.com/dapr/components-contrib/pull/2863). It is used here in the unit test as a realistic example.

**Review by yaron2 — APPROVED** [review]
